common/pkgfiles/referencedfilelist.cpp

```diff
@@ -126,13 +126,38 @@ void splitDerivedDfsLocation(const char *address, StringBuffer &cluster, StringB
     prefix.append(basePrefix);
 }
 
+void splitDerivedDfsLocationOrRemote(const char *address, StringBuffer &cluster, StringBuffer &ip, StringBuffer &prefix,
+                                    const char *defaultCluster, const char *baseCluster, const char *baseIP, const char *basePrefix,
+                                    const char *currentRemoteStorage, StringBuffer& effectiveRemoteStorage, const char *baseRemoteStorage)
+{
+    if (!isEmptyString(address) && !isEmptyString(currentRemoteStorage))
```

Comment:
You're right that it shouldn't happen at the same level.  I think you do support what Ananth wants which is to for example have the packagemap define a remoteStorage, but some packages inside it point to a daliIP?

Reply:
I believe it should support that case, but I need to confirm with a test.

Reply:
I've tested behavior where a CLI setting of either `daliip` or `remote-storage` is overridden by either `daliip` or `remoteStorage` set in the packagemap file. Also, those properties defined in the file have a precedence over each other- a property defined in an element will override the value defined in its ancestors.